app/frontend/components/pages/delegations/delegateInput.tsx

```diff
@@ -27,48 +27,75 @@ const StakePoolLabel = ({
+    validStakepoolDataProvider.getPoolInfoByPoolHash(fieldValue) ||
+    validStakepoolDataProvider.getPoolInfoByTicker(fieldValue)
+  if (pool) return {poolHash: pool.poolHash, error: null}
+
+  const hasTickerMapping = validStakepoolDataProvider.hasTickerMapping
+  const isTickerString = fieldValue.length <= 5 && fieldValue.toUpperCase() === fieldValue
+  const poolHash = null
+  if (!hasTickerMapping && isTickerString) return {poolHash, error: {code: 'TickerSearchDisabled'}}
+  return {poolHash, error: {code: 'InvalidStakepoolIdentifier', params: {hasTickerMapping}}}
+}
+
 interface Props {
   poolRecommendation: PoolRecommendation
   pool: Stakepool
   validStakepoolDataProvider: StakepoolDataProvider
   updateStakePoolIdentifier: (poolHash: string, validationError?: any) => void
+  resetStakePoolIndentifier: () => void
 }
 
 const DelegateInput = ({
   poolRecommendation,
   pool,
   validStakepoolDataProvider,
   updateStakePoolIdentifier,
+  resetStakePoolIndentifier,
 }: Props): h.JSX.Element => {
-  useEffect(() => {
-    const poolHash = poolRecommendation?.recommendedPoolHash || pool?.poolHash
-    if (poolHash) {
-      updateStakePoolIdentifier(poolHash)
-    }
-  }, [pool, poolRecommendation, updateStakePoolIdentifier])
-
-  const [fieldValue, setFieldValue] = useState(
-    poolRecommendation?.recommendedPoolHash || pool?.poolHash || ''
-  )
+  const [fieldValue, setFieldValue] = useState('')
   const [isTicker, setIsTicker] = useState(false)
   const [isPoolHash, setIsPoolHash] = useState(!!fieldValue)
 
+  useEffect(() => {
+    const recommendedPoolHash = poolRecommendation?.recommendedPoolHash || pool?.poolHash
+    if (recommendedPoolHash) {
+      const {poolHash, error} = validateInput(recommendedPoolHash, validStakepoolDataProvider)
+      updateStakePoolIdentifier(poolHash, error)
+      setFieldValue(recommendedPoolHash)
+    }
+  }, [pool, poolRecommendation, validStakepoolDataProvider, updateStakePoolIdentifier])
+
   const handleOnInput = (event: any): void => {
     const fieldValue: string = event?.target?.value
     const isTicker = fieldValue && !!validStakepoolDataProvider.getPoolInfoByTicker(fieldValue)
     const isPoolHash = fieldValue && !!validStakepoolDataProvider.getPoolInfoByPoolHash(fieldValue)
     setFieldValue(fieldValue)
     setIsTicker(isTicker)
     setIsPoolHash(isPoolHash)
-    const poolHash = isTicker
-      ? validStakepoolDataProvider.getPoolInfoByTicker(fieldValue).poolHash
-      : fieldValue
-    const hasTickerMapping = validStakepoolDataProvider.hasTickerMapping
-    const error =
-      poolHash === '' || validStakepoolDataProvider.getPoolInfoByPoolHash(poolHash)
-        ? null
-        : {code: 'InvalidStakepoolIdentifier', params: {hasTickerMapping}}
-    updateStakePoolIdentifier(poolHash, error)
+
+    if (fieldValue) {
+      const {poolHash, error} = validateInput(fieldValue, validStakepoolDataProvider)
+      updateStakePoolIdentifier(poolHash, error)
+    } else {
```

Comment:
Is there a situation when `onInput` event is triggered and fieldValue is undefined or null? Just asking if this actually does something...

Reply:
Yes, this splits logic when the fields is empty or not empty.

Reply:
:thinking: Interesting.. thanks :+1: